File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        vector<int> prefix;
        vector<int> suffix(nums.size());
        int sum = 0;
        for(int i=0; i<nums.size(); i++){
            prefix.push_back(sum);
            sum +=nums[i];
        }

        int sum_suff = 0;
        for(int j=nums.size()-1; j>=0; j--){
            suffix[j] = sum_suff;
            sum_suff += nums[j];
        }

        int n = 0;
        int m = 0;
        while(n<nums.size()){
            if(prefix[n] == suffix[n]) return n;
            n++;
        }
        return -1;
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:

        prefix = []
        suffix = [0]*len(nums)

        sum_pre = 0
        for i in range(len(nums)):
            prefix.append(sum_pre)
            sum_pre += nums[i]
        
        sum_suff = 0
        
        for i in range(len(nums)-1,-1,-1):
            suffix[i] = sum_suff
            sum_suff += nums[i]

        n =0
        while n< len(nums):
            if(prefix[n]==suffix[n]): return n
            n +=1
        return -1
        